### ISSUE TITLE
Fossilize (vulkan shaders) and I/O load issues

### ISSUE THREAD

**Issue by SimplyCorbett** [ValveSoftware/Fossilize#182]

**Comment by kakra** [comment]
This has mainly be discussed here: https://github.com/ValveSoftware/Fossilize/issues/99

The behavior you observe is due to the introduction of watching IO PSI in the kernel which is global across all drives. You could try turning the kernel PSI feature off: Add `psi=0` to your kernel command line (https://facebookmicrosites.github.io/psi/docs/overview). It may be possible that your kernel defaults to on.

This change actually included PSI support which fixed desktop stalls for most users: https://github.com/ValveSoftware/Fossilize/commit/200b19c319e2872415d74b5d3479e1624d748bc6

It was introduced because shader compilation is not actually a CPU-only thing, it also involves a lot of inefficient IO (it's actually not much IO but it is pretty random in the driver caches and thus inefficient, especially on btrfs).

**Comment by HansKristian-Work** [comment]
Yes, Fossilize has to go out of its way to not make other stuff go slow on the system, especially anything related to IO since we can quickly swarm IO caches when 10+ threads hammer out shader caches with non-ideal access patterns.

**Comment by kakra** [comment]
@HansKristian-Work It may be possible that the dirty pages watcher is a bit too aggressive: If copying large files, dirty data is expected. Maybe it should watch PSI only, and when there is no PSI, it should fall back to a less aggressive dirty pages watcher? Or just make that less aggressive in general?

Personally, I don't care if it pauses when running in the background. It's probably the foreground mode when people would care about it. That said, it works perfectly fine for me in background mode since that change back then - no issues whatsoever. Not sure if the Steam client already uses the control channel to actually switch to aggressive mode when running in foreground. But then again, when such a control option is implemented but Steam doesn't use it, this is probably a Steam bug, not a Fossilize bug.

**Comment by kakra** [comment]
> since we can quickly swarm IO caches when 10+ threads hammer out shader caches with non-ideal access patterns

That's actually the point here even when copying data on another drive... Fossilize needs to get out of the way of other software using the page cache.
